frontend/src/api/k8s/__tests__/notebooks.spec.ts

```diff
@@ -0,0 +1,968 @@
+import {
+  k8sCreateResource,
+  k8sGetResource,
+  k8sPatchResource,
+  k8sListResource,
+  k8sUpdateResource,
+  k8sDeleteResource,
+  K8sStatus,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import * as _ from 'lodash-es';
+import { NotebookKind } from '~/k8sTypes';
+import { Toleration } from '~/types';
+import { mockNotebookK8sResource } from '~/__mocks__/mockNotebookK8sResource';
+import { mockK8sResourceList } from '~/__mocks__/mockK8sResourceList';
+import { mock200Status } from '~/__mocks__/mockK8sStatus';
+import { mockRoleBindingK8sResource } from '~/__mocks__/mockRoleBindingK8sResource';
+import { mockStartNotebookData } from '~/__mocks__/mockStartNotebookData';
+
+import {
+  assembleNotebook,
+  createNotebook,
+  getNotebook,
+  getNotebooks,
+  stopNotebook,
+  startNotebook,
+  updateNotebook,
+  deleteNotebook,
+  attachNotebookSecret,
+  replaceNotebookSecret,
+  attachNotebookPVC,
+  removeNotebookPVC,
+  removeNotebookSecret,
+  getStopPatch,
+  startPatch,
+} from '~/api/k8s/notebooks';
+
+import {
+  createElyraServiceAccountRoleBinding,
+  getPipelineVolumePatch,
+  getPipelineVolumeMountPatch,
+  ELYRA_VOLUME_NAME,
+} from '~/concepts/pipelines/elyra/utils';
+
+import { TolerationChanges, getTolerationPatch } from '~/utilities/tolerations';
+import { NotebookModel } from '~/api/models';
+import { getImageStreamDisplayName } from '~/pages/projects/screens/spawner/spawnerUtils';
+
+jest.mock('@openshift/dynamic-plugin-sdk-utils', () => ({
+  k8sCreateResource: jest.fn(),
+  k8sDeleteResource: jest.fn(),
+  k8sGetResource: jest.fn(),
+  k8sListResource: jest.fn(),
+  k8sPatchResource: jest.fn(),
+  k8sUpdateResource: jest.fn(),
+}));
+
+jest.mock('~/concepts/pipelines/elyra/utils', () => ({
+  createElyraServiceAccountRoleBinding: jest.fn(),
+  getPipelineVolumePatch: jest.fn(),
+  getPipelineVolumeMountPatch: jest.fn(),
+  getElyraVolume: jest.fn(() => ({
+    name: 'elyra-dsp-details',
+    secret: {
+      secretName: 'ds-pipeline-config',
+    },
+  })),
+  getElyraVolumeMount: jest.fn(() => ({
+    name: 'elyra-dsp-details',
+    mountPath: '/opt/app-root/src/data',
+  })),
+  getshmVolume: jest.fn(() => ({
+    name: 'shm',
+    emptyDir: { medium: 'Memory', sizeLimit: '64M' },
+  })),
+  getshmVolumeMount: jest.fn(() => ({
+    name: 'shm',
+    mountPath: '/dev/shm',
+  })),
+  ELYRA_VOLUME_NAME: 'elyra-dsp-details',
+}));
```

Comment:
This mock is concerning. It ends up returning a lot of the same values. Why even bother mocking these?

Reply:
@christianvogt these mock was to cover the following https://github.com/opendatahub-io/odh-dashboard/blob/31b8ce30a4798836ce40740d4b4ff04363de8c37/frontend/src/api/k8s/notebooks.ts#L70-L90 lines because `getElyraVolume()` `getElyraVolumeMount()` etc was returning `volume.name is undefined`

Reply:
Sounds like you've created the problem by mocking more than you require. Use `jest.requireActual` to mixin the original implementation then mock only the essential functions that you require.

Reply:
@christianvogt thanks for suggestion! updated accordingly.